packages/endo/src/parse.js

```diff
@@ -0,0 +1,115 @@
+/* global StaticModuleRecord */
+
+import { parseRequires } from "./parse-requires.js";
+import { parseExtension } from "./extension.js";
+
+const { entries, freeze, fromEntries } = Object;
+
+// q, as in quote, for enquoting strings in error messages.
+const q = JSON.stringify;
+
+// TODO: parsers should accept bytes and perhaps even content-type for
+// verification.
+
+export const parseMjs = (source, location) => {
+  return new StaticModuleRecord(source, location);
+};
+
+export const parseCjs = (source, location) => {
+  if (typeof source !== "string") {
+    throw new TypeError(
+      `Cannot create CommonJS static module record, module source must be a string, got ${source}`
+    );
+  }
+  if (typeof location !== "string") {
+    throw new TypeError(
+      `Cannot create CommonJS static module record, module location must be a string, got ${location}`
+    );
+  }
+
+  const imports = parseRequires(source, location);
+  const execute = (exports, compartment, resolvedImports) => {
```

Comment:
CommonJS has always supported cyclic dependencies in this fashion. Two modules can mutually require as long as they lazy-bind properties of the exports returned by `require`, and as long as they use `exports.property =` assignment instead of `module.exports =` replacement. The `module.exports` property was not part of the original CommonJS, although it has become dominant usage in Node.js. The `module` object originally existed in CommonJS to fulfill the same function as `import.meta`.